### DISCUSSION TITLE
Barricade with  grenade script weird behaviour

### DISCUSSION THREAD

**Discussion by DerpyHoowes** [SmartlyDressedGames/Unturned-3.x-Community#4013] [Q&A] [answered]
So I was making a time fuse charge barricade, which would explode with some delay after you place it. Just a grenade script attached to the barricade, nothing fancy. And it worked fine for several days, until today. I didnt change anything about the barricade, but now the explosion effect spawns not in the transform of the barricade, but instead seemingly in the map's center, under the ground.
Does it have to do anything with the way barricades load? If I remember correctly, there's something special about the barricade hierarchy in the game, to make it consume less RAM or something. Or was there any quick patch, that wasnt announced that changed something?

I guess it could be this way from the beginning, and I just didnt notice it because maybe I tested it only on vehicles (and they have their own explosion effect, so I could have missed the fact that the charge itself has no explosion effect), but Im pretty sure I did place it on the ground, and it did make explosion effect spawn properly, until now.

I know that the scripts arent meant to be used that way, but still Im rather curious why this is happening.

**Comment by DerpyHoowes** [comment, marked as answer]
Turns out it was indeed because of the barricade pooling. Solved it by adding Eligible_For_Pooling false to the asset.

**Reply by SDGNelson** [reply]
Note that using the script to destroy the barricade will probably have some unintended side effects like leaking the barricade instance responsible for spawning it.
